FormRecognizer/MLOpsTips.md

```diff
@@ -0,0 +1,392 @@
+      "1099": "328e73d7-04ec-4da8-b0a2-631201d7fa48",
+      "1040X": "70a798ac-b955-4c04-b47c-40ddba700c87"
+    }
+  }
+}
+```
+
+And the target mapping file is:
+
+```json
+{
+  "dev": {
+    "westus": {
+      "1040": "a2190780-1e47-4602-bf06-00b70548c63e",
+      "1040V": "89818d7e-a84b-40c3-ab39-0c2e83f28dbf",
+      "1040ES": "af1b07ea-3d53-4a2e-a629-f4c996e4ba96",
+      "1099R": "937359c8-a40e-475b-9e8d-1891655ae394",
+      "1099": "328e73d7-04ec-4da8-b0a2-631201d7fa48"
+    }
+  },
+  "staging": {
+    "eastus2": {
+      "1040": "442d6021-b6db-4ba9-a239-66e4110474c1",
+      "1040V": "70a469b2-c962-47bc-9a1f-aadd7055b375",
+      "1040ES": "218095cd-fb10-4f03-a7a7-677cdf217fc5",
+      "1099R": "043cd574-38c9-4d5f-ad4a-b05f50f07b4f",
+      "1099": "e10dad9a-cd03-4e96-a550-321cee950356"
+    }
+  }
+}
+```
+
+In the target mapping file, the dev->westus object is the snapshot of the source environment when the previous migration happened.
+So the script will compare dev->westus in source with dev->westus in target, and detect that the id for `1040V` has changed and `1040X` has been added since the previous migration.
+
+Thus the script will know to migrate `1040V` and `1040X` ids to the target env.
+Once the migration has completed, the target mapping file will be updated with the newly copied model ids, and the dev->westus object in the target file will be replaced with the dev->westus object in the source file, as you need to snapshot the source environment's state during this migration.
+So the next time the migration script runs, it will be able to detect new or retrained layouts.
+
+Like training, the script needs to track the ids that were replaced or removed from the target mapping file and delete these models from the target FR instance to save space.
+
+While the target mapping file has multiple environment keys, the inference service for that environment can be set to target a specific path, i.e. the staging service will know to index into staging->eastus2 to find its mappings.
+These mapping files can be chained together for as many environments as needed, meaning the updated target file can be used as the source file for the next environment.
+Saving multiple environment's mapping in a single mapping file also helps with auditing, as you can see what the state of both environments were during the previous migration, and the script can be set to continuously build up this environment migration recorded if desired.
+
+## Model Migration Script
+
+This is the core logic of the model migration script, without the downloading and uploading of the various mapping files.
+You need to get the mapping files into python dicts as well as get your FR keys, endpoints, and resource_ids in the `envConfigs` class for the script to work.
+Currently, the script does not delete models from target that were removed in the source mapping file, but there is a `#TODO` where that logic should be added.
+
```

Comment:
@mover333 - Thank you, can you please add the Microsoft and license directive here please